src/protocol/graphics.ts

```diff
@@ -286,6 +286,10 @@ export function setupGraphics(store: UIStore) {
 }
 
 export async function repaint(force = false) {
+  const recordingTarget = await ThreadFront.recordingTargetWaiter.promise;
+  if (recordingTarget === "node") {
```

Comment:
Small nit, but would prefer if the waiters are private and expose a cleaner interface...

```js
if (await ThreadFront.getTarget() == "node") {
  return;
}
```